tests/interpreters/test_cf_interpreter.py

```diff
@@ -0,0 +1,69 @@
+from xdsl.builder import Builder, ImplicitBuilder
+from xdsl.dialects import arith, cf, func
+from xdsl.dialects.builtin import ModuleOp, i32
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters.arith import ArithFunctions
+from xdsl.interpreters.cf import CfFunctions
+from xdsl.interpreters.func import FuncFunctions
+from xdsl.ir.core import Block, Region
+
+
+def triangle_fn(n: int) -> int:
+    result = 0
+    i = 0
+    while i <= n:
```

Comment:
This can be done better?

Reply:
Is that a question or an assertion? the whole function is `n * (n-1) // 2` so can be done better indeed :)